pkg/providers/snow/fetch.go

```diff
@@ -50,3 +51,15 @@ func oldWorkerMachineTemplate(ctx context.Context, kubeclient kubernetes.Client,
 	}
 	return mt, nil
 }
+
+func oldCredentialsSecret(ctx context.Context, kubeClient kubernetes.Client, clusterSpec *cluster.Spec) (*v1.Secret, error) {
+	secret := &v1.Secret{}
+	err := kubeClient.Get(ctx, CredentialsSecretName(clusterSpec), constants.EksaSystemNamespace, secret)
+	if apierrors.IsNotFound(err) {
+		return nil, nil
```

Comment:
This is a bit confusing, is it better to handle this not found case in the caller of this function?

Reply:
the caller function intentionally handles not found error  separately so that this method can treat not found differently than other errors.